learnr_bank/learnr_set2.Rmd

```diff
@@ -0,0 +1,468 @@
+  + Baseline chunk code: 
+        example_array <- c(5, 78, -3, 11, 6, 0)
+        example_array * 5
+  + **Hint:** When modifying the code be careful not to delete the definition of `example_array`.
+  + Solution:   example_array / 10
+  
+```{r array_q6, exercise = TRUE}
+example_array <- c(5, 78, -3, 11, 6, 0)
+example_array * 5
+```      
+
+<div id= "array_q6-hint">
+**Hint:** When modifying the code be careful not to delete the definition of `example_array`.
+</div>
+
+7. We can also perform logical operations in a vectorized context. The code below asks if all values in `example_array` are greater than 0. Run this code and examine its output to be sure you understand it. Then, modify the code to instead ask if every item is *less than or equal to 0**.
+  + Baseline chunk code: 
+        example_array <- c(5, 78, -3, 1, 6, 0)
+        example_array > 0
+  + **Hint 1:** When modifying the code be careful not to delete the definition of `example_array`.
+  + **Hint 2:** This type of output would be referred to as a *logical array*.
+  + Solution:   example_array <= 0
+
+```{r array_q7, exercise = TRUE}
+example_array <- c(5, 78, -3, 1, 6, 0)
+example_array > 0
+```  
+
+```{r array_q7-hint-1}
+#When modifying the code be careful not to delete the definition of `example_array`.
+```
+
+```{r array_q7-hint-2}
+#This type of output would be referred to as a *logical array*.
+```
+
+8. In fact, logical values TRUE and FALSE evaluate to numbers 1 and 0, respectively. Use the function `sum()` (adds up values in an array) to calculate the sum of the output from `example_array > 0`. 
+  + **Hint 1:** Don't forget to define `example_array` again here.
+  + **Hint 2:** You can provide the full expression `example_array > 0` to the function `sum()`, or save that logical expression's output to a variable, and use the variable as the argument to `sum()`.
+  + Solution:  example_array <- c(5, 78, -3, 1, 6, 0)
+               sum(example_array > 0)
+               
+```{r array_q8, exercise = TRUE}
+
+```                 
+
+```{r array_q8-hint-1}
+#Don't forget to define `example_array` again here.
+```
+
+```{r array_q8-hint-2}
+#You can provide the full expression `example_array > 0` to the function `sum()`, or save that logical expression's output to a variable, and use the variable as the argument to `sum()`.
+```
+
+### Set 2: Answer Key
+
+``` {r, array_q1-answer}
+#Solution to Question 1
+array_numbers <- c(1,2,3,4,5)
+array_numbers
+```
+
+``` {r, array_q2-answer}
+#Solution to Question 2
+array_numbers2 <- 1:5
+array_numbers2
+```
+
+``` {r, array_q3-answer}
+#Solution to Question 3
+even_numbers <- c(2,4,6,8,10)
+even_numbers
+```
+
+``` {r, array_q4-answer}
+#Solution to Question 4
+even_numbers <- seq(2, 10, 2)
+even_numbers
+```
+
+``` {r, array_q5-answer}
+#Solution to Question 5
+mul3 <- seq(0, 999, 3)
+length(mul3)
+```
+
+``` {r, array_q6-answer}
+#Solution to Question 6
+example_array <- c(5, 78, -3, 11, 6, 0)
```

Comment:
I needed to define the example array in the answer chunk for 6 and 7. Otherwise, it would not run.

Reply:
Yup, this sounds right - have to keep redefining since all chunks are separate.